File: java/app/src/main/java/com/fatchickensoftware/books/BookDetailActivity.java
package com.fatchickensoftware.books;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.NavUtils;
import android.support.v4.content.ContextCompat;
import android.support.v7.app.AppCompatActivity;
import android.view.MenuItem;

import com.fatchickensoftware.books.databinding.ActivityBookDetailBinding;

import static com.fatchickensoftware.books.BookDetailFragment.INTENT_VOLUME;

/**
 * An activity representing a single Item detail screen. This
 * activity is only used on narrow width devices. On tablet-size devices,
 * item details are presented side-by-side with a list of items
 * in a {@link BookListActivity}.
 */
public class BookDetailActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        ActivityBookDetailBinding binding = ActivityBookDetailBinding.inflate(getLayoutInflater(), null);
        binding.setBackgroundColor(ContextCompat.getColor(this, R.color.transparent));
        setContentView(binding.getRoot());

        setSupportActionBar(binding.detailToolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setDisplayShowTitleEnabled(false);

        Intent intent = getIntent();
        Volume volume = intent != null ? (Volume) intent.getSerializableExtra(INTENT_VOLUME) : null;
        if (volume != null) {
            binding.setVolumeInfo(volume.volumeInfo);
            binding.setTextColor(ContextCompat.getColor(this, R.color.white));

            Fragment fragment = BookDetailFragment.createInstance(volume);
            getSupportFragmentManager().beginTransaction()
                    .add(R.id.item_detail_container, fragment)
                    .commit();
        }
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();
        if (id == android.R.id.home) {
            NavUtils.navigateUpTo(this, new Intent(this, BookListActivity.class));
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    public static Intent createInstance(Context context, Volume volume) {
        Intent intent = new Intent(context, BookDetailActivity.class);
        intent.putExtra(BookDetailFragment.INTENT_VOLUME, volume);
        return intent;
    }
}

File: kotlin/app/src/main/java/com/fatchickensoftware/books/BookDetailActivity.kt
package com.fatchickensoftware.books

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v4.content.ContextCompat
import android.support.v7.app.AppCompatActivity
import android.view.MenuItem
import com.fatchickensoftware.books.databinding.ActivityBookDetailBinding

/**
 * An activity representing a single Item detail screen. This
 * activity is only used on narrow width devices. On tablet-size devices,
 * item details are presented side-by-side with a list of items
 * in a [BookListActivity].
 */
class BookDetailActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val binding = ActivityBookDetailBinding.inflate(layoutInflater, null)
        binding.backgroundColor = ContextCompat.getColor(this, R.color.transparent)
        setContentView(binding.root)

        setSupportActionBar(binding.detailToolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.setDisplayShowTitleEnabled(false)

        (intent?.getSerializableExtra(BookDetailFragment.INTENT_VOLUME) as? Volume)?.let { volume ->
            binding.volumeInfo = volume.volumeInfo
            binding.textColor = ContextCompat.getColor(this, R.color.white)

            val fragment = BookDetailFragment.createInstance(volume)
            supportFragmentManager.beginTransaction()
                .add(R.id.item_detail_container, fragment)
                .commit()
        }
    }

    override fun onOptionsItemSelected(item: MenuItem) =
        when (item.itemId) {
            android.R.id.home -> {
                navigateUpTo(Intent(this, BookListActivity::class.java))
                true
            }
            else -> super.onOptionsItemSelected(item)
        }

    companion object {
        fun createInstance(context: Context, volume: Volume): Intent =
            Intent(context, BookDetailActivity::class.java).apply {
                putExtra(BookDetailFragment.INTENT_VOLUME, volume)
            }
    }
}
